File: python/socketd/socketd/transport/core/FrameIoHandler.py
from abc import ABC, abstractmethod
from typing import TypeVar, Callable

from socketd.transport.core.ChannelAssistant import ChannelAssistant
from socketd.transport.core.ChannelInternal import ChannelInternal
from socketd.transport.core.Frame import Frame

S = TypeVar("S")

# 帧输入输出处理器 # 为 TrafficLimiter 提供支持
class FrameIoHandler(ABC):
    # 发送帧处理
    @abstractmethod
    async def send_frame_handle(self, channel: ChannelInternal, frame: Frame, channelAssistant: ChannelAssistant[S],
                          target: S, completionHandler:Callable[[bool, Exception], None]):
        ...

    # 接收帧处理
    @abstractmethod
    async def reve_frame_handle(self, channel: ChannelInternal, frame: Frame):
        ...

File: javascript/socketd/src/socketd/transport/core/FrameIoHandler.ts
import type {ChannelInternal} from "./Channel";
import {Frame} from "./Frame";
import {ChannelAssistant} from "./ChannelAssistant";
import {IoBiConsumer} from "./Typealias";

/**
 * 帧输入输出处理器（为 TrafficLimiter 提供支持）
 * */
export interface FrameIoHandler {
    /**
     * 发送帧
     *
     * @param channel          通道
     * @param frame            帧
     * @param channelAssistant 通道助理
     * @param target           发送目标
     */
    sendFrameHandle<S>(channel: ChannelInternal, frame: Frame, channelAssistant: ChannelAssistant<S>, target: S, completionHandler:IoBiConsumer<Boolean, Error|any>);

    /**
     * 接收帧
     *
     * @param channel 通道
     * @param frame   帧
     */
    reveFrameHandle(channel: ChannelInternal, frame: Frame);
}